diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68973eb7..c91a87dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,7 +318,7 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
         -Wno-sign-compare -Wno-unused-local-typedefs -Wno-unknown-pragmas)
 ## Clang
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Wpedantic -Wunreachable-code -fsized-deallocation
+    add_compile_options(-Wall -Wextra -Wpedantic -Wunreachable-code -fno-sized-deallocation
         -Wstrict-aliasing -Wno-sign-compare -Wno-unused-local-typedef -Wno-unknown-pragmas)
 endif()
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9195f051..6c141e8e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -111,7 +111,7 @@ if(ENABLE_FREESASA)
 endif()
 
 # target: unittests
-add_executable(unittests unittests.cpp ${tsts})
+add_executable(unittests EXCLUDE_FROM_ALL unittests.cpp ${tsts})
 target_link_libraries(unittests libfaunus)
 target_compile_definitions(unittests PRIVATE SPDLOG_COMPILED_LIB)
 add_test(
@@ -463,9 +463,9 @@ install(FILES
     ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide.yml
     ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide.out.json
     ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide.state.json
-    ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide_molecular.yml
-    ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide_molecular.out.json
-    ${EXAMPLES_DIR}/calciumhydroxide/calciumhydroxide_molecular.state.json
+    ${EXAMPLES_DIR}/calciumhydroxide_molecular/calciumhydroxide_molecular.yml
+    ${EXAMPLES_DIR}/calciumhydroxide_molecular/calciumhydroxide_molecular.out.json
+    ${EXAMPLES_DIR}/calciumhydroxide_molecular/calciumhydroxide_molecular.state.json
     ${EXAMPLES_DIR}/water/ewald.yml
     DESTINATION share/faunus/examples)
 
